### PR TITLE
Button: Making Button variants themable by using createComponent instead of wrappers

### PR DESCRIPTION
**Pull request by khmakoto**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

This PR fixes themability in `Button` variants by making them use `createComponent` instead of being wrappers on the `Button` component. The approach taken here was to create them in such a way so that `Button` does not appear in their hierarchy (`MenuButton` and `SplitButton` still have `Button` in their hierarchy, this just applies to the variants under the `ButtonVariants` folder). The styles have been tweaked so that no visual regressions are present using this approach.

This PR also adds some high contrast mode fixes to a number of  the variants as well as `SplitButton` that can be seen below:

__Before:__

image

__After:__

image

#### Focus areas to test

`BaseButton` was removed as part of this PR since it's dependent on the work in #9436. I'll add the `Do Not Merge` label until that other PR is merged.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9605)

### CODE REVIEW

File: apps/vr-tests/src/stories/NewButton.stories.tsx [hunk 1]
```diff
@@ -192,23 +192,23 @@ storiesOf('ActionButton', module)
...
+//         .snapshot('hovered', { cropTo: '.testWrapper' })
+//         .mouseDown('.ms-Button')
+//         .snapshot('pressed', { cropTo: '.testWrapper' })
+//         .end()}
+//     >
+//       {story()}
+//     </Screener>
+//   ))
+//   .addStory('Default', () => <BaseButton {...baseProps} />, { rtl: true })
+//   .addStory('Disabled', () => <BaseButton disabled {...baseProps} />);
```
Comment (JasonGore) [inline apps/vr-tests/src/stories/NewButton.stories.tsx]: Hmm what's the story here?

Comment (khmakoto) [reply apps/vr-tests/src/stories/NewButton.stories.tsx]: I think I wrote that down in the PR description, but basically I'm removing `BaseButton` on this PR just to get it out for review of the other parts while we wait on getting #9436 in. Then I can merge that in here and get `BaseButton` again.

Comment (JasonGore) [reply apps/vr-tests/src/stories/NewButton.stories.tsx]: I did read the description, I just didn't see the association with this test. ;)

File: apps/perf-test/src/scenarios/BaseButtonNew.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import * as React from 'react';
-import { BaseButton } from '@uifabric/experiments';
+import { Button } from '@uifabric/experiments';
 
-const scenario = <BaseButton text="I am a button" />;
+const scenario = <Button content="I am a button" />;
```
[the remark this answers is not in the document]
Comment (khmakoto) [reply apps/perf-test/src/scenarios/BaseButtonNew.tsx]: I'll also change this when #9436 merges.

**Pull request merged**